adr/0007-quarkus-workshop-structure.adoc

```diff
@@ -0,0 +1,77 @@
+At the time, we anticipated additional workshops, leading us to establish a dedicated structure within a single repository: Quarkus Workshops (https://github.com/quarkusio/quarkus-workshops).
+
+The initial (and still existing) structure was straightforward:
+
+[source]
+----
+.
+├── README.md
+└── quarkus-workshop-super-heroes/
+    ├── dist
+    ├── docs
+    ├── super-heroes
+    ├── README.adoc
+    └── pom.xml
+----
+
+Although this structure was meant to support multiple workshops, only the "Quarkus Superheroes" workshop was added.
+Instead of separate workshops, we expanded this initial workshop with additional steps and features.
+
+As we now develop new workshops on various topics, we face limitations with the single repository structure.
+For example, the Quarkus LangChain4J workshop was created separately to demonstrate Quarkus LangChain4J usage, yet it isn’t integrated into the main workshop repository.
+Additionally, having a single repository complicates using GitHub Pages for documentation.
+
+Given the current and future workshops, it’s essential to reconsider the structure to allow easier management and discoverability of each workshop.
+
+== Proposed New Structure
+
+Our experience shows that hosting all workshops in one repository isn’t optimal. We propose a new structure as follows:
+
+1. Each workshop will be hosted in its own repository.
+This simplifies management, avoids conflicts in `README` and documentation setup, and improves workshop discoverability.
+2. Naming convention: Each workshop repository should follow the format `quarkus-workshop-<topic>`, where `<topic>` represents the workshop subject (e.g., `quarkus-workshop-superheroes`, `quarkus-workshop-langchain4j`).
+3. Documentation should be hosted with GitHub Pages in each repository, making each workshop more accessible.
+
+== Considered Options
+
+=== Option 1: Continue with the current single-repository approach
+
+This would mean keeping all workshops under the existing repository.
+However, as observed, this approach has not met expectations and makes workshop management more challenging.
+
+=== Option 2: Create a separate organization for workshops
+
+A dedicated organization could host all workshops, offering a single access point.
+However, this approach could reduce discoverability, but would not use the Quarkus organization’s CI resources.
+CI resource usage is minor, as workshops are not frequently updated.
+
+== Consequences
+
+=== Positive
+
+* Simplified workshop management.
+* Greater autonomy for workshop maintainers.
+* Consolidation of workshops previously hosted in separate repositories.
+
+=== Negative
+
+* Lack of a central place to list all workshops. This could be mitigated by creating a dedicated page on the Quarkus website.
+* Potential CI resource shortage as each workshop repository uses _quarkusio_ organization CI resources.
+However, this is unlikely to be a significant issue, as, generally, workshops don't use much CI resources.
```

Comment:
Well, if you are going to have tests in your workshop that can run so that versions stay up to date, then this might start to become true....

Reply:
if these does start take significant resources we can move them to a new org

Reply:
Sure, but let's tackle one problem at a time. Moving them to a separate organization also has downsides.

Reply:
@cescoffier if it was unclear - I was just stating that if it becomes a problem it is fairly easy to move individual repos somewhere else. so we agree..

though 2.5hrs feels crazy ....what is that build doing ;)

Reply:
... because having them in a separate org would be letting the tail wag the dog, and sacrificing discoverability to accommodate a technical limitation of GitHub runners.

Reply:
> though 2.5hrs feels crazy ....what is that build doing ;)

It's a static site that needs to host variations for each OS, each combination of features, etc (for example, a workshop host could share the URL https://quarkus.io/quarkus-workshops/super-heroes/variants/os-all-ai-false-azure-false-cli-false-container-false-contract-testing-true-extension-false-kubernetes-false-messaging-true-native-false-observability-false/spine.html with their participants). So it builds each variation, and the combinatorics multiply exponentially.The alternative is to generate dynamically, but that means standing up a back end and move away from GitHub pages, but that has a lot of downside.